### PR TITLE
feat(render): add render framework

### PR DESCRIPTION
**Pull request by SirXuNian**
add engine render frame work

### CODE REVIEW

File: src/engine/gfx/renderJob/config/RenderLayer.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/**
+ * 渲染类型标签
```
Comment (hellmor) [inline src/engine/gfx/renderJob/config/RenderLayer.ts:2]: style for code

File: src/engine/gfx/renderJob/jobs/RendererJob.ts [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+import { PickFire } from '../../../io/PickFire';
+import { ShadowLightsCollect } from '../collect/ShadowLightsCollect';
+import { GBufferFrame } from '../frame/GBufferFrame';
+import { RendererMap } from './RenderMap';
+import { ColorPassRenderer } from '../passRenderer/color/ColorPassRenderer';
+import { Graphic3D } from '../passRenderer/graphic/Graphic3DRender';
+import { ReflectionProbeRenderer } from '../passRenderer/probe/ReflectionProbeRenderer';
+
+/**
+ * 渲染器任务类
```
Comment (Codeboy-cn) [inline src/engine/gfx/renderJob/jobs/RendererJob.ts:29]: Please use english or remove it.

File: src/engine/gfx/renderJob/passRenderer/RenderContext.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        this.rendererPassStates = [];
+    }
+
+    public clean() {
+        this.rendererPassStates.length = 0;
+        GPUContext.cleanCache();
+    }
+
+    /**
+     * 继续一个连续的 renderer pass state
```
Comment (Codeboy-cn) [inline src/engine/gfx/renderJob/passRenderer/RenderContext.ts:24]: Please use english or remove it.

**Review by Codeboy-cn — CHANGES_REQUESTED** [review]

File: src/engine/gfx/renderJob/jobs/RendererJob.ts (hunk 2 above)
Comment (hellmor) [inline src/engine/gfx/renderJob/jobs/RendererJob.ts:29]: Modify the language of annotation text

File: src/engine/gfx/renderJob/jobs/RendererJob.ts [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+    public reflectionProbeRenderer: ReflectionProbeRenderer;
+
+    /**
+     * @internal
+     */
+    public pauseRender: boolean = false;
+    public pickFire: PickFire;
+
+    /**
+   * 图形渲染器(线、矩形等)
```
Comment (hellmor) [inline src/engine/gfx/renderJob/jobs/RendererJob.ts:94]: Modify the language of annotation text

File: src/engine/gfx/renderJob/passRenderer/RenderContext.ts (hunk 3 above)
Comment (hellmor) [inline src/engine/gfx/renderJob/passRenderer/RenderContext.ts:24]: Modify the language of annotation text

File: src/engine/gfx/renderJob/state/RendererMask.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
+﻿/**
+ * RenderMask 状态值
```
Comment (hellmor) [inline src/engine/gfx/renderJob/state/RendererMask.ts:2]: Modify the language of annotation text

**Review by Codeboy-cn — APPROVED** [review]

File: src/engine/gfx/renderJob/passRenderer/RendererBase.ts [hunk 6]
```diff
@@ -0,0 +1,227 @@
...
+        return [];
+    }
+
+    protected recordRenderBundleNode(view: View3D, encoder, nodes: RenderNode[]) {
+        GPUContext.bindCamera(encoder, view.camera);
+        GPUContext.bindGeometryBuffer(encoder, nodes[0].geometry);
+        for (let i = 0; i < nodes.length; ++i) {
+            let renderNode = nodes[i];
+            let matrixIndex = renderNode.transform.worldMatrix.index;
+            if (!renderNode.transform.enable) continue;
```
Comment (hellmor) [inline src/engine/gfx/renderJob/passRenderer/RendererBase.ts:181]: wrap line, and enclose in parentheses

File: src/engine/gfx/renderJob/passRenderer/RendererBase.ts [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+            renderNode.recordRenderPass2(view, this._rendererType, this.rendererPassState, this.clusterLightingRender, encoder);
+        }
+    }
+
+    protected drawRenderNodes(view: View3D, encoder: GPURenderPassEncoder, command: GPUCommandEncoder, nodes: RenderNode[], occlusionSystem: OcclusionSystem) {
+        GPUContext.bindCamera(encoder, view.camera);
+        for (let i = Engine3D.setting.render.drawOpMin; i < Math.min(nodes.length, Engine3D.setting.render.drawOpMax); ++i) {
+            let renderNode = nodes[i];
+            if (!occlusionSystem.renderCommitTesting(view.camera, renderNode)) continue;
+            if (!renderNode.transform.enable) continue;
```
Comment (hellmor) [inline src/engine/gfx/renderJob/passRenderer/RendererBase.ts:191]: wrap line, and enclose in parentheses

**Review by hellmor — APPROVED** [review]

**Review by hellmor — APPROVED** [review]

**Pull request merged**